Source/missiles.cpp

```diff
@@ -395,25 +389,13 @@ void CheckMissileCol(Missile &missile, int minDamage, int maxDamage, bool isDama
 	int my = position.y;
 
 	bool isMonsterHit = false;
-	const int mid = dMonster[mx][my];
-	if (missile._micaster != TARGET_BOTH && !missile.IsTrap()) {
-		if (missile._micaster == TARGET_MONSTERS) {
-			if (mid != 0 && (mid > 0 || Monsters[abs(mid) - 1]._mmode == MonsterMode::Petrified)) {
-				isMonsterHit = MonsterMHit(missile._misource, abs(mid) - 1, minDamage, maxDamage, missile._midist, missile._mitype, isDamageShifted);
-			}
-		} else {
-			Monster &attackingMonster = Monsters[missile._misource];
-			if ((attackingMonster._mFlags & MFLAG_TARGETS_MONSTER) != 0
-			    && mid > 0
-			    && (Monsters[mid - 1]._mFlags & MFLAG_GOLEM) != 0)
-				isMonsterHit = MonsterTrapHit(mid - 1, minDamage, maxDamage, missile._midist, missile._mitype, isDamageShifted);
-		}
-	} else {
-		if (mid > 0) {
-			if (missile._micaster == TARGET_BOTH)
-				isMonsterHit = MonsterMHit(missile._misource, mid - 1, minDamage, maxDamage, missile._midist, missile._mitype, isDamageShifted);
-			else
-				isMonsterHit = MonsterTrapHit(mid - 1, minDamage, maxDamage, missile._midist, missile._mitype, isDamageShifted);
+	int mid = dMonster[mx][my];
+	if (mid > 0 || (mid != 0 && Monsters[abs(mid) - 1]._mmode == MonsterMode::Petrified)) {
+		mid = abs(mid) - 1;
+		if (missile.IsTrap() || (missile._micaster == TARGET_PLAYERS && (Monsters[missile._misource]._mFlags & MFLAG_TARGETS_MONSTER) != 0 && Monsters[missile._misource]._menemy == mid)) {

```

Comment:
It's a fair consideration, but counter argument would be:
- It's going to be very rare
- Why is the arrow hitting monsters that they are angry at. We might as well argue that it should hit any monster.

Reply:
Sorry I forgot what a problem is here. It is not a missile cannot hit non-golem monster. But as originally found:
missiles originating from berserking monster cannot hit non-golem (that means also non-berserking) monsters.

while MvM *friendly fire* could be some mod feature (2nd bullet), for 1st bullet I think even if it occurs rarely you can get it fixed without any cost by changing this
`Monsters[missile._misource]._menemy == mid`
into this:
` ((Monsters[mid - 1]._mFlags & MFLAG_GOLEM) != 0 || (Monsters[missile._misource]._mFlags & MFLAG_BERSERK) != 0))`
as originally fixed in my PR.

imagine this again: you have berserked monster. As soon as he is berserked - he start shooting things - player or other monsters - whoever closer.
Now his arrow (or other missile) will travel to hit targeted monster, miss it, and fly through monsters behind it and not even try to hit monsters behind it.
Compare to player missile - he shoot a missile targeted at some monster. Assume Missile will miss first monster on a way, it will travel further and try to hit monster behind.
Consider berserking monster that casts lightning (all monsters behind target would be safe).
That would even look weird. As only targeted monster would be vulnerable.
In my opinion there is no reason to left this condition like that.

Actually For 2nd bullet - there is no problem already, and no problem with my proposed solution. Since berserking monster CAN hit each-other - as they have MFLAG_GOLEM flag.

Reply:
If you open a PR with this I can merge it.

Reply:
> I don't feel strongly either way. If you open a PR with this I can merge it.

Ok I will do it later. When I will be back from bicycle trip :)
btw, this could be avoided if you would look at my PRs waiting for review ;)
(I am still convinced it should not have vanilla tag as bugfix for petrified vs FW, maybe only as tag meaning it is change vs vanilla. This PR clearly solves HF issues for me, as HF introduced real MvM with berserkers.)

Reply:
There are over 100 pending PRs, and plenty of them conflict with things I or others are working on. I try my best but there is a lot to see to and only so much time to do it in.

Reply:
> There are over 100 pending PRs, and plenty of them conflict with things I or others are working on. I try my best but there is a lot to see to and only so much time to do it in.

I understand, but there are really few small ones too.
Also some (any) feedback would be better than nothing - like which area/change/PR being worked it conflicts with.
Maybe then I could help too in those areas.

Reply:
Maybe you could join us on the chat where we talk about development

Reply:
I am on discord already, if you mean `#development` channel

Reply:
you must have a different name there. Maybe you can introduce yourself